changelog.d/712.feature.md

```diff
@@ -0,0 +1 @@
+Added the `return` expression as per RFC 7496.
```

Comment:
Can we add more details here?

Reply:
Agreed. Maybe something like:

```suggestion
Added the `return` expression as per [RFC 7496](https://github.com/vectordotdev/vector/blob/4671ccbf0a6359ef8b752fa99fae9eb9c60fdee5/rfcs/2023-02-08-7496-vrl-return.md). This expression can be used to terminate the VRL program early while still emitting a value.
```

This reminds me that we'll need to update the docs too as a separate PR to https://github.com/vectordotdev/vector

Reply:
Who is the changelog geared at? Should the addition of returns to the typedef or other implementation details also mentioned or is the suggestion from @jszwedko good enough?

Reply:
The audience of the changelog is users of VRL and Vector so the technical bits can be left for the commit changelog (which has VRL contributors as its audience).